### PR TITLE
[POI] add ShouldInterrupt for SelectPointOfInterestPrompt, SelectActionPrompt

### LINKED ISSUE #2212: [POI] Support scenario where user can search for different POIs than results shown
## User Story
As ...
I want ...
so that ...

## Acceptance Criteria 
*List your requirements.*
- [ ] ...
- [ ] ...
- [ ] ...

## Resources
*Provide any screenshots or conversation mockups to support your request.*
*Learn how to mock up bot conversations with [.chat files](http://aka.ms/bfchatfile).*

Support conversations like:
User: find a coffee shop
Skill: here are some coffee shops
User: find a pharmacy
Skill: here are some pharmacies
User: find a KEYWORD
Skill: here are some KEYWORD

## Notes
*Provide any additional information.*
The POI Skill should be more flexible in continually searching for new POIs if a user asks without being stuck on what they uttered previously.

### PR DESCRIPTION
**Pull request by xieofxie**
### Purpose
*What is the context of this pull request? Why is it being done?*

Close #2212 

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

* Set ShouldInterrupt when it is not recognized by prompt and recognized by PointOfInterestLuis in InterruptablePromptValidator for SelectPointOfInterestPrompt and SelectActionPrompt
  - If run luis first, some option would be treated as valid intent
  - Although it depends on skill state, it is always cleared when begin, so I think it is fine
* In ContinueDialogAsync, return DialogTurnStatus.Empty when ShouldInterrupt since RouterDialog will call RouteAsync when receives it
  - It depends on logic of RouterDialog
* Fix selectAction

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

* Add test RouteToPointOfInterestAndInterruptTest

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

After #2266 for better utterances

#### General
- [x] I have commented my code, particularly in hard-to-understand areas	
- [x] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [ ] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [ ] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [ ] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [ ] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

**Comment by xieofxie** [comment]
@ryanlengel a somehow non-standard approach

File: skills/src/csharp/Directory.Build.props [hunk 1]
```diff
@@ -3,6 +3,12 @@
     <CodeAnalysisRuleSet>$(SolutionDir)..\..\..\VirtualAssistant.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
 
+  <PropertyGroup>
```
Comment (ryanisgrig) [inline skills/src/csharp/Directory.Build.props]: Does this still need myget?

Comment (xieofxie) [reply skills/src/csharp/Directory.Build.props]: Updated~ They are no longer needed.
But they are now blocked by #2435 as PRValidation is changed

**Comment by ryanisgrig** [comment]
Please rebase against next to capture the refactoring changes

**Comment by xieofxie** [comment]
Rebased~

**Review by ryanisgrig — APPROVED** [review]

**Review by ryanisgrig — APPROVED** [review]

**Review by lauren-mills — APPROVED** [review]

**Pull request merged**